apps/server/src/modules/learnroom/controller/course.controller.ts

```diff
@@ -47,4 +74,28 @@ export class CourseController {
 		});
 		return new StreamableFile(result);
 	}
+
+	@Post('import')
+	@UseInterceptors(FileInterceptor('file'))
+	@ApiOperation({ summary: 'Imports a course from a Common Cartridge file.' })
+	@ApiConsumes('multipart/form-data')
+	@ApiBody({ type: CourseImportBodyParams, required: true })
+	@ApiCreatedResponse({ description: 'Course was successfully imported.' })
+	@ApiBadRequestResponse({ description: 'Request data has invalid format.' })
+	@ApiInternalServerErrorResponse({ description: 'Internal server error.' })
+	public async importCourse(
+		@CurrentUser() currentUser: ICurrentUser,
+		@UploadedFile(
+			new ParseFilePipe({
+				validators: [
+					new MaxFileSizeValidator({ maxSize: config().MAX_FILE_SIZE }),
```

Comment:
This can not work no pod cache can handle this. 
If i look to the code, i think the best case is that CommonCartridge is a one micro service that is deployed seperatly and use filesstorage, nextcloud, or core server if it is needed. 

course/import is also not the best place ...if i read this url i expect that it is our internal course import/export. 
But CommonCartridge is a completly seperated topic.

Reply:
Regarding the file size, it's the file size i got after asking in the DevOps + SRE Open RocketChanel (18.01.2024).

The micro service implementation is planned for the future and not scope of this ticket. Documentation for the design of a micro service would be helpfull.

What is a better place for the import? A Suggestion would be helpful.

Reply:
When it is outsource to a additional micro service than you must move all the code to a addtional folder with new controller api endpoints and so on. Then it is also fine. But it should be avoid that any operations are executed on production until it is seperated. Otherwise the server pods can be killed by out of memory.